### DISCUSSION TITLE
Do we still need to add `app.set('trust proxy', 1);`?

### DISCUSSION THREAD

**Discussion by liamli99** [nfriedly/express-rate-limit#473] [Q&A] [answered]
Hi!

In the older documentation, this line of code `app.set('trust proxy', 1)` is suggested when we want to host the application on a cloud service which uses reverse proxy.

However, in the latest documentation: https://www.npmjs.com/package/express-rate-limit, this line of code has been removed, do we still need to add `app.set('trust proxy', 1)` in the beginning of our project if we use the latest version of `express-rate-limit`?

**Comment by gamemaker1** [comment]
Hi @liamli99,

You only need to use the trust proxy if you are facing issues with the limiter behind a reverse proxy. See [this guide](https://express-rate-limit.mintlify.app/guides/troubleshooting-proxy-issues#the-global-limiter-problem) for a more detailed explanation.

**Reply by liamli99** [reply]
Hi @gamemaker1, 

Thank you for your explanation! Is there any specific error message coming up when facing this issue? The fact is my application is deployed successfully on a cloud service without using trust proxy, and I don't know if it is behind a reverse proxy.

**Reply by gamemaker1** [reply, marked as answer]
One symptom to look for is that the rate limiter applies globally, i.e., any requests that are made count towards a single global limit.

This occurs when the trust proxy setting isn't used properly, and the reverse proxy IP is used as the key for every single request, instead of the actual client's IP. See the page I linked in the earlier comment regarding the same.

**Reply by liamli99** [reply]
Thanks! This is really helpful!

**Comment by nfriedly** [comment]
One reason it's not in the readme any more is that the library now automatically detects common proxy misconfiguration issues and logs an error, for example:

https://github.com/express-rate-limit/express-rate-limit/blob/ff3702e4d92c2ab93022d8f66e0c3786c3b8d799/source/validations.ts#L124-L134

That will log an error that links to https://express-rate-limit.mintlify.app/reference/error-codes#err-erl-unexpected-x-forwarded-for
